R/makephyloseqFromTreeSummarizedExperiment.R

```diff
@@ -1,219 +1,135 @@
+    args = list()
+    # Gets the abundance data from assay, and converts it to otu_table
+    otu_table <- as.matrix(assay(x, assay.type))
+    otu_table <- phyloseq::otu_table(otu_table, taxa_are_rows = TRUE)
+    # Adds to the list
+    args[["otu_table"]] <- otu_table
+
+    # If rowData includes information
+    if(!( length(rowData(x)[,taxonomyRanks(x)]) == 0 ||
+            is.null((rowData(x)[,taxonomyRanks(x)])) )){
+        # Converts taxonomy table to characters if it's not already
+        rowData(x) <- DataFrame(lapply(rowData(x), as.character))
+        # Gets the taxonomic data from rowData, and converts it to tax_table
+        tax_table <- as.matrix(rowData(x)[,taxonomyRanks(x),drop=FALSE])
+        tax_table <- phyloseq::tax_table(tax_table)
+        # Adds to the list
+        args[["tax_table"]] <- tax_table
+    }
         
-        # phyloseq object requires nonduplicated rownames. If there are 
-        # duplicated rownames, they are converted so that they are unique
-        if( any(duplicated(rownames(x))) ){
-            rownames(x) <- getTaxonomyLabels(x)
-        }
-        # List of arguments
-        args = list()
-        # Gets the abundance data from assay, and converts it to otu_table
-        otu_table <- as.matrix(assay(x, assay.type))
-        otu_table <- phyloseq::otu_table(otu_table, taxa_are_rows = TRUE)
+    # If colData includes information
+    if(!( length(colData(x)) == 0 || is.null(ncol(colData(x))) )){
+        # Gets the feature_data from colData and converts it to sample_data
+        sample_data <- as.data.frame(colData(x))
+        sample_data <- phyloseq::sample_data(sample_data)
         # Adds to the list
-        args[["otu_table"]] <- otu_table
-
-        # If rowData includes information
-        if(!( length(rowData(x)[,taxonomyRanks(x)]) == 0 ||
-              is.null((rowData(x)[,taxonomyRanks(x)])) )){
-            # Converts taxonomy table to characters if it's not already
-            rowData(x) <- DataFrame(lapply(rowData(x), as.character))
-            # Gets the taxonomic data from rowData, and converts it to tax_table
-            tax_table <- as.matrix(rowData(x)[,taxonomyRanks(x),drop=FALSE])
-            tax_table <- phyloseq::tax_table(tax_table)
-            # Adds to the list
-            args[["tax_table"]] <- tax_table
-        }
+        args[["sample_data"]] <- sample_data
+    }
         
-        # If colData includes information
-        if(!( length(colData(x)) == 0 || is.null(ncol(colData(x))) )){
-            # Gets the feature_data from colData and converts it to sample_data
-            sample_data <- as.data.frame(colData(x))
-            sample_data <- phyloseq::sample_data(sample_data)
-            # Adds to the list
-            args[["sample_data"]] <- sample_data
-        }
+    # Creates a phyloseq object
+    phyloseq <- do.call(phyloseq::phyloseq, args)
+    return(phyloseq)
+}
+
+
+.make_phyloseq_from_TreeSE <- function(x, tree.name = "phylo", ...){
```

Comment:
For users still using old name, modify this line to:

.make_phyloseq_from_TreeSE <- function(x, tree.name = tree_name, tree_name = "phylo", ...){